rust/src/dns/log.rs

```diff
@@ -664,12 +741,19 @@ pub extern "C" fn SCDnsLogJsonAnswer(
 }
 
 #[no_mangle]
-pub extern "C" fn SCDnsLogAnswerEnabled(tx: &mut DNSTransaction, flags: u64) -> bool {
-    if let Some(response) = &tx.response {
-        for query in &response.queries {
-            if dns_log_rrtype_enabled(query.rrtype, flags) {
-                return true;
-            }
+pub extern "C" fn SCDnsLogEnabled(tx: &mut DNSTransaction, flags: u64) -> bool {
+    let message = if let Some(request) = &tx.request {
+        request
+    } else if let Some(response) = &tx.response {
+        response
+    } else {
+        // Should be unreachable...
+        return false;
```

Comment:
Debug validation.. In fact, the DNSTransaction could be restructured with a type of request/response, and a message body giving us only 2 branches.